--- ibmtpm974.orig/src/Implementation.h	2017-03-21 22:41:16.000000000 +0800
+++ ibmtpm974/src/Implementation.h	2018-02-28 10:05:16.939524955 +0800
tpm2: Make RADIX_BITS dependent on openssl #defines

Make the #define RADIX_BITS dependent on #defines set by the
OpenSSL library: THIRTY_TWO_BIT and SIXTY_FOUR_BIT_LONG
Patch code is picked from Stefan Berger's libtpms.
See commit: https://github.com/stefanberger/libtpms/commit/33c1f3c3c40e307663359a3505fc52cdaf311fd8

Signed-off-by: Liu Qun <qunliu@zyhx-group.com>
@@ -304,7 +304,16 @@
 /* From Vendor-Specific: Table 7 - Defines for Implementation Values */
 #define  FIELD_UPGRADE_IMPLEMENTED      NO
 #ifdef TPM_POSIX
-#define  RADIX_BITS                     64	/* kgold */
+# include <openssl/bn.h>
+# ifdef THIRTY_TWO_BIT
+#  define RADIX_BITS                     32
+# endif
+# ifdef SIXTY_FOUR_BIT_LONG
+#  define RADIX_BITS                     64
+# endif
+# ifndef RADIX_BITS
+#  error Need to determine RADIX_BITS value
+# endif
 #endif
 #ifdef TPM_WINDOWS
 #define  RADIX_BITS                     32	/* kgold */
