--- ibmtpm974.orig/src/Implementation.h	2017-03-21 22:41:16.000000000 +0800
+++ ibmtpm974.mine/src/Implementation.h	2018-02-28 10:05:16.939524955 +0800
@@ -304,7 +304,7 @@
 /* From Vendor-Specific: Table 7 - Defines for Implementation Values */
 #define  FIELD_UPGRADE_IMPLEMENTED      NO
 #ifdef TPM_POSIX
-#define  RADIX_BITS                     64	/* kgold */
+#define  RADIX_BITS                     32
 #endif
 #ifdef TPM_WINDOWS
 #define  RADIX_BITS                     32	/* kgold */
